packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -486,10 +497,11 @@ export class ConnectionManager implements IConnectionManager {
 	private set_readonlyPermissions(
 		newReadonlyValue: boolean,
 		oldReadonlyValue: boolean | undefined,
+		reason?: string,
```

Comment:
consider using IConnectionStateChangeReason here, so errors can be tracked as well